ros2_batch_job/__main__.py

```diff
@@ -505,19 +500,23 @@ def need_package_from_pipy(pkg_name):
             # to ensure that the build type specific package is installed
             job.run(
                 ['"%s"' % job.python, '-m', 'pip', 'uninstall', '-y'] +
-                [f'cryptography{pip_cryptography_version}', 'lxml', 'numpy'], shell=True)
+                ['cryptography', 'lxml', 'numpy'], shell=True)
+
         with open('constraints.txt', 'w') as outfp:
-            outfp.write('empy < 4\n')
-            outfp.write('flake8 < 5.0.0\n')
-            outfp.write('setuptools==59.6.0\n')
-            outfp.write('pytest==6.2.5\n')
+            outfp.write('\n'.join(constraints))
 
         pip_cmd = ['"%s"' % job.python, '-m', 'pip', 'install', '-c', 'constraints.txt', '-U']
         if sys.platform == 'win32':
             # Force reinstall so all dependencies are in virtual environment
             # On Windows since we switch between the debug and non-debug
             # interpreter all packages need to be reinstalled too
             pip_cmd.append('--force-reinstall')
+        else:
+            # We need to use --user so that certain packages (like flake8-blind-except)
+            # can successfully build on RHEL-8 (Humble).
```

Comment:
In that case does it make sense to restrict the addition of this flag to the platform that requires it?

Reply:
> In that case does it make sense to restrict the addition of this flag to the platform that requires it?

Yeah, but unfortunately here in this script, we don't have that information.  I'm currently using `humble` as a proxy for it, but that is probably a bad idea overall.  I'll think about how to do this better, we may have to add a new `--platform` flag.